fix(find): respect spaces when query has no whitespace

shreeMahadikGit opened the pull request:
Fixes this issue:

https://github.com/mozilla/pdf.js/issues/20225

shreeMahadikGit committed:
fix(find): respect spaces when query has no whitespace

calixteman reviewed (changes requested):
Unfortunately, it won't work with a pdf containing "Hello. World" and a query equals to "o. w" (fyi it's ok in Acrobat).
I think a fix could be to add optional white spaces around group of punctuation signs which induces to not have [ ]* between consecutive punctuation signs.
It'd lead to update `SPECIAL_CHARS_REG_EXP` but you've to take care to the case of `.` and ` ?`.

shreeMahadikGit committed:
Handleded for this test case - o. w

shreeMahadikGit commented:
Handled the Suggested test case. Also Attaching the PDF with the test case added:

[Morse (1) (1).pdf](https://github.com/user-attachments/files/22943390/Morse.1.1.pdf)

calixteman commented:
@shreeMahadikGit do you plan to work on this patch or not ?

shreeMahadikGit commented:
Hii @calixteman,

First of all, apologies for the late reply. I found a few more issues with the solution I implemented.

As you can see in the attached screenshot, the search isn’t matching the correct characters. I’ll need to do some additional work on my solution to address this.

I have refactored the code according to your comments and removed most of the unnecessary parts as well, but I’ve discovered a few more issues in the current implementation that need to be fixed.

**If you think someone else can address this more quickly, please feel free to reassign the issue.**

<img width="907" height="768" alt="Screenshot 2025-10-31 at 12 37 24 AM" src="https://github.com/user-attachments/assets/d9fe4de2-c9f2-4a18-b076-79446efca374" />
